src/helpers/getPrettierPath.js

```diff
@@ -1,14 +1,38 @@
 // @flow
 const path = require('path');
+const { sync: execaSync } = require('execa');
 const { findCached } = require('atom-linter');
+const { memoize } = require('lodash');
 const globalModules = require('global-modules');
 const yarnGlobalModules = require('yarn-global-modules')();
 const { findCachedFromFilePath } = require('./general');
 
 const PRETTIER_INDEX_PATH = path.join('node_modules', 'prettier', 'index.js');
 
+// global-prefix package (which is used by global-modules) fails
+// to locate the directory if Node and NPM are installed via NVM
+// https://github.com/jonschlinkert/global-prefix/issues/19
+// The function below applies a fallback search mechanism,
+// which relies on calling 'npm get prefix'
+const getFallbackGlobalModulesPath = memoize(() => {
+  try {
```

Comment:
Which line specifically is it that can throw an exception and why? From looking at `execa`, it looks like it can throw errors if you call a bad command, for example. However, we probably don't want to swallow errors in that case like this? 

Do we expect that sometimes the user won't have `npm` on the path or something? If so, we should do a pre-emptive guard for that by executing `which`?

For example, here is roughly the same approach but no try/catch so that we don't swallow unexpected errors:

```js
const npmIsAvailable = () => !_.isEmpty(execaSync("which", ["npm"]).stdout);

const npmPrefix = () => execaSync("npm", ["get", "prefix"]).stdout;

const resolveGlobalNodeModulesFromPrefix = prefix =>
  process.platform === "win32" ||
  process.env.OSTYPE === "msys" ||
  process.env.OSTYPE === "cygwin"
    ? path.resolve(prefix, "node_modules")
    : path.resolve(prefix, "lib/node_modules");

const resolveGlobalNodeModulesWithNpm = () => _.flow(npmPrefix, resolveGlobalNodeModulesFromPrefix);

const getFallbackGlobalModulesPath = _.memoize(() => _.cond([[npmIsAvailable, resolveGlobalNodeModulesWithNpm]]));
```

Reply:
> Which line specifically is it that can throw an exception and why?

I'm generally trying to guard the execa call just in case, since the signature of that function does allow for an exception to be thrown. I can't think of a concrete example, but it still feels safer with `try` / `catch`. I would prefer this solution over calling `which` because `which` may (?) not be available on Windows and also because it's another sync CLI call that takes time.

Your solution implies calling `execaSync` every time, even when fallback is not needed. This is what I was tryign to avoid by implementing `getFallbackGlobalModulesPath` the way I did. I'm totally happy to copy-paste your code though because I don't think that an extra CLI call is that expensive. WDYT?

Reply:
Well, the whole thing does get memoized at the end in the code sample I put so there should only be those two calls to execa and then that's it.

But, the fact that `which` may not be present in Windows is concerning. I wasn't thinking about that. Let's merge for now and then we can always address later if we find this becomes a problem (which it probably won't).